### PR TITLE
Fix `backend-api-spec-update.yml`

### PR DESCRIPTION
**Pull request by nulls**
### What's done:
- set fetch-depth: 0 <-- this fixes pull issue
- moved git configuration to earlier step <-- author is required to rebase\merge changes from master

---
Not tested case:
1. we have opened PR
2. we changes some endpoint
3. the workflow regenerates `backend-api-docs.json`
4. the workflow tries to create PR but it's already existed...

### CODE REVIEW

File: .github/workflows/backend-api-spec-update.yml [hunk 1]
```diff
@@ -19,21 +19,27 @@ jobs:
       - uses: actions/checkout@v3
         with:
           ref: 'infra/update-backend-api-spec'
+          fetch-depth: 0
```
Comment (kgevorkyan) [inline .github/workflows/backend-api-spec-update.yml:22]: could you add comment about this code

File: .github/workflows/backend-api-spec-update.yml [hunk 2]
```diff
@@ -19,21 +19,27 @@ jobs:
...
 
       - name: Set up Java
         uses: actions/setup-java@v3
         with:
           java-version: 17
           distribution: temurin
+
+      - name: Set up Git
```
Comment (kgevorkyan) [inline .github/workflows/backend-api-spec-update.yml:30]: how did it worked previously without this 🤦‍♂️

**Review by kgevorkyan — APPROVED** [review]

**Comment by kgevorkyan** [comment]
> 4. the workflow tries to create PR but it's already existed...

I think in worst case, this step will just fail, but ``git pull` is first, so all should be okay at the end

File: .github/workflows/backend-api-spec-update.yml (hunk 2 above)
Comment (nulls) [reply .github/workflows/backend-api-spec-update.yml:30]: I'm not sure, maybe `infra/update-backend-api-spec` doesn't have any changes\commits during git pull

**Comment by nulls** [comment]
> > 4. the workflow tries to create PR but it's already existed...
> 
> I think in worst case, this step will just fail, but `git pull` is first, so all should be okay at the end

yeap, but it breaks automatization, but will return to this question if it will gets more problems.

Note: automatization is still broken, since PR and commits are done by bot user and builds are not triggered automatically

File: .github/workflows/backend-api-spec-update.yml (hunk 1 above)
Comment (nulls) [inline .github/workflows/backend-api-spec-update.yml:22]:
```suggestion
          fetch-depth: 0 # we need a full history to make rebase with master
```

Comment (nulls) [reply .github/workflows/backend-api-spec-update.yml]: done

Comment (kgevorkyan) [reply .github/workflows/backend-api-spec-update.yml]: good idea in aim to do fast commit via github 😀

**Pull request merged**